docs/configuration.md

```diff
@@ -693,6 +693,64 @@ value: <tmpl_string>
 [ short: <boolean> | default = slack_config.short_fields ]
 ```
 
+## `<sns_configs>`
+```yaml
+# Whether or not to notify about resolved alerts.
+[ send_resolved: <boolean> | default = false ]
+
+# The SNS API URL i.e. https://sns.us-east-2.amazonaws.com.
+#  If not specified, the SNS API URL from the SNS SDK will be used.
+[api_url: <tmpl_string>]
+
+# Configures AWS's Signature Verification 4 signing process to sign requests.
+sigv4:
+  [ <sigv4_config> ]
+  
+# SNS topic ARN, i.e. arn:aws:sns:us-east-2:698519295917:My-Topic
+# If you don't specify this value, you must specify a value for the phone_number or target_arn.
+# If you are using a FIFO SNS topic you should set a message group interval longer than 5 minutes
+# to prevent messages with the same group key being deduplicated by the SNS default deduplication window
+[ topic_arn: <tmpl_string>]
+
+# Subject line when the message is delivered to email endpoints.
+[ subject: <tmpl_string> | default = '{{ template "sns.default.subject" .}}' ] 
+
+# Phone number if message is delivered via SMS in E.164 format.
+# If you don't specify this value, you must specify a value for the topic_arn or target_arn.
+[ phone_number: <tmpl_string> ] 
+
+# The  mobile platform endpoint ARN if message is delivered via mobile notifications.
+# If you don't specify this value, you must specify a value for the topic_arn or phone_number.
+[ target_arn: <tmpl_string> ] 
+
+# The message content of the SNS notification.
+[ message: <tmpl_string> | default = '{{ template "sns.default.message" .}}' ] 
+
+# SNS message attributes.
+attributes: 
+  [key : value]
+
+# The HTTP client's configuration.
+[ http_config: <http_config> | default = global.http_config ]
```

Comment:
Should we have a global sigv4 config? anfd global api keys etc?

Reply:
I added add a global sigV4 config, for each field in the config the hierarchy in order of precedence is: 
1) ` sns.sigv4.< access_key | secret_key | region | profile | roleARN > `
2) `global.sigv4.< access_key | secret_key | region | profile | roleARN > `
3) environmental variables if the field is empty in both configs listed above

I think this is a little complicated, but it does follow the same pattern we use for the other global api keys (Slack, PagerDuty, VictorOps). I'm not sure if you have any ideas on how to make it clearer, or if this is a non-issue. The other option I can think of is flattening the global sigv4 object into `global.sigv4_access_key, global.sigv4_secret_key, ... `

Reply:
We decided to only use the local sns.sigv4 config, and not add a global.